akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -311,7 +310,13 @@ final case class Drop[T](count: Long) extends SimpleLinearGraphStage[T] {
       if (left > 0) {
         left -= 1
         pull(in)
-      } else push(out, grab(in))
+      } else {
+        push(out, grab(in))
+        // After dropping N elements, the following behavior is just pushing
+        setHandler(in, new InHandler {
```

Comment:
hm, this is a tradeoff between an extra allocation and a branch which is totally predictable. I vote for the branch here.